### PR TITLE
XS-3032: Defensive improvements

### PR DESCRIPTION
**Pull request by chrislococo-wf**
#### Changes:

- Moved logic into _default_dimension_mismatch function. 
- Update unit tests.
- Encapsulated the main plugin loop in a generic exception class to avoid unhandled bubbling up into Arelle. 

Please review: @hefischer  @andrewperkins-wf

### CODE REVIEW

File: dqc_us_rules/dqc_us_0041.py [hunk 1]
```diff
@@ -269,22 +269,27 @@ def _check_relationship_exists(rel):
...
     dimensions
 
-    :param default_dimension: dimension to test if it is a usgaap default
-        dimension
-    :type: str
-    :param usgaap_default_dimensions: list of usgaap taxonomies default
-        dimensions
-    :type: list
+    :param relation: dimension to test if it is a usgaap default dimension.
+    :type: :class:`~arelle.ModelDtsObject.ModelRelationship`
```
Comment (brianmyers-wf) [inline dqc_us_rules/dqc_us_0041.py]: for your type docstrings, are you not including the parameter name anymore?

Comment (chrislococo-wf) [inline dqc_us_rules/dqc_us_0041.py]: I have evolved beyond the need for types :neckbeard:

**Comment by brianmyers-wf** [comment]
+1

File: dqc_us_rules/__init__.py [hunk 2]
```diff
@@ -12,12 +12,17 @@ def run_checks(val):
...
+            if ((plugin.__file__ is not None and
+                 plugin.__file__.find('__init__.py') == -1 and
+                 hasattr(plugin, '__pluginInfo__'))):
+
+                func = plugin.__pluginInfo__['Validate.XBRL.Finally']
+                func(val)
+        except Exception:
+            # This is an overly generic error catch, but it will hopefully
+            # be able to be pared down in the future.
+            val.modelXbrl.error(sys.exc_info())
```
Comment (andrewperkins-wf) [inline dqc_us_rules/__init__.py]: This should be more than just the exc_info().  It needs to have an error key, and a few other things as well.  I will attempt to find an example of this.

**Comment by andrewperkins-wf** [comment]
looks good outside of the message, I will look at the right way to record that.

File: dqc_us_rules/__init__.py (hunk 2 above)
Comment (andrewperkins-wf) [inline dqc_us_rules/__init__.py]: [Example](https://github.com/Arelle/Arelle/blob/master/arelle/Validate.py#L74)

**Comment by andrewperkins-wf** [comment]
pass complete

**Comment by brianmyers-wf** [comment]
This should be closed as it is now duplicated by: https://github.com/DataQualityCommittee/dqc_us_rules/pull/98

**Pull request closed without merging**